app/controllers/NextUpdatesController.scala

```diff
@@ -76,17 +76,18 @@ class NextUpdatesController @Inject()(NoNextUpdatesView: NoNextUpdates,
           case (_, true) =>
             auditNextUpdates(user, isAgent, origin)
 
-            val optOutSetup = for {
-              checks <- optOutService.getNextUpdatesQuarterlyReportingContentChecks
-              optOutOneYearViewModel <- optOutService.nextUpdatesPageOptOutViewModel()
-              _ <- optOutService.resetSavedIntent()
-            } yield Ok(nextUpdatesOptOutView(viewModel, optOutOneYearViewModel, checks, backUrl.url, isAgent, origin))
-
-            optOutSetup recover {
-              case ex =>
-                Logger("application").error(s"Unexpected future failed error, ${ex.getMessage}")
-                errorHandler.showInternalServerError()
+            val optOutSetup = optOutService.getNextUpdatesQuarterlyReportingContentChecks.flatMap { checks =>
```

Comment:
Can we just keep the call for the `checks`  inside the for comprehension? I think that as long as the whole comprehension is within braces, then the `recoverWith()` can apply to that.

Reply:
Done

Reply:
Cool.